compiler/quilt/test/test_push.py

```diff
@@ -59,11 +67,70 @@ def test_push(self):
         # Push it again; this time, we're verifying that there are no s3 uploads.
         command.push('foo/bar')
 
-    def _mock_put_package(self, package, pkg_hash, upload_urls):
+    def test_push_subpackage(self):
+        mydir = os.path.dirname(__file__)
+        build_path = os.path.join(mydir, './build_simple.yml')
+        command.build('foo/bar', build_path)
+
+        _, pkgroot = store.PackageStore.find_package(None, 'foo', 'bar')
+        contents = pkgroot.children['foo']
+
+        all_hashes = set(find_object_hashes(contents))
+        upload_urls = {
+            blob_hash: dict(
+                head="https://example.com/head/{owner}/{hash}".format(owner='foo', hash=blob_hash),
+                put="https://example.com/put/{owner}/{hash}".format(owner='foo', hash=blob_hash)
+            ) for blob_hash in all_hashes
+        }
+
+        for blob_hash in all_hashes:
+            urls = upload_urls[blob_hash]
+
+            self.requests_mock.add(responses.HEAD, urls['head'], status=404)
+            self.requests_mock.add(responses.PUT, urls['put'])
+
+        self._mock_update_package('foo/bar', 'foo', contents, upload_urls)
+
+        # Push a subpackage.
+        command.push('foo/bar/foo')
```

Comment:
this _should_ fail without `is_pubilc=True`?

Reply:
at least for a mocked up "free" user.

Reply:
Well, there's no server... It's all mocked by the `callback1` and `callback2` methods below. I could make them fail without `is_public=True`, but that wouldn't test anything other than "does the client pass the is_public parameter to the server?", so really trivial.